com.trekglobal.idempiere.rest.api/src/com/trekglobal/idempiere/rest/api/v1/resource/impl/UploadResourceImpl.java

```diff
@@ -531,17 +653,32 @@ public void reassembleFile(MRestUpload upload, String finalFileName, List<MRestU
 				throw new IOException("SHA-256 mismatch for final file. Expected: " + upload.getREST_SHA256() + ", Calculated: " + sha256);
 			}
             
-            MArchive archive = new Query(Env.getCtx(), MArchive.Table_Name, "AD_Table_ID=? AND Record_ID=?", upload.get_TrxName())
-					.setParameters(MRestUpload.Table_ID, upload.getREST_Upload_ID()).first();
-            if (archive == null) {
-				archive = new MArchive(Env.getCtx(), 0, upload.get_TrxName());
-				archive.setAD_Table_ID(MRestUpload.Table_ID);
-				archive.setRecord_ID(upload.getREST_Upload_ID());
-				archive.setRecord_UU(upload.getREST_Upload_UU());
-			}
-            archive.setName(upload.getFileName());
-            archive.setBinaryData(baos.toByteArray());
-            archive.saveEx();
+            if (upload.getREST_UploadLocation().equals(MRestUpload.REST_UPLOADLOCATION_Image)) {
```

Comment:
This is not correct, the destination to copy to should not depends on the initial upload location. For e.g, the initial upload location is archive and now we wants to attach it to the pass in record.